Search2D.java

```diff
@@ -0,0 +1,25 @@
+
```

Comment:
class Solution {
    //Time O(m+n)
    //Space O(1)
    public boolean searchMatrix(int[][] matrix, int target) {
        if(matrix == null || matrix.length == 0)
        {
            return false;
        }
        int row = matrix.length-1 , col = 0;
        while(row >=0 && col < matrix[0].length)
        {
            if(matrix[row][col] == target)
            {
                return true;
            }
            else if(matrix[row][col] > target)
            {
                row--; // All the numbers to its clearly greater than the current element so move upwards
            }
            else
            {
                col++;
            }
        }
        return false;
    }
}